test/functional/runtime/autoload/provider/python3_spec.lua

```diff
@@ -0,0 +1,93 @@
+do
+  local proc =
+    io.popen([[python3 -c 'import neovim, sys; sys.stdout.write("ok")' 2> /dev/null]])
+  if proc:read() ~= 'ok' then
+    -- Don't run these tests if python3 is not available
+    return
+  end
+end
+
+
+local helpers = require('test.functional.helpers')
+local eval, command, feed = helpers.eval, helpers.command, helpers.feed
+local eq, clear, insert = helpers.eq, helpers.clear, helpers.insert
+local expect = helpers.expect
+
+
+describe('python3 commands and functions', function()
+  before_each(function()
+    clear()
+    command('python3 import vim')
+  end)
+
+  describe('feature test', function()
```

Comment:
@Shougo I didn't mean to suggest that you should nest `it()`. But the nested `describes()` in this case are also pointless. 

Here is the exact suggestion:

```lua
describe('python3 commands and functions', function()
  before_each(function()
    clear()
    command('python3 import vim')
  end)

  it('feature test', function()
    eq(1, eval('has("python3")'))
  end)

  it('python3_execute', function()
    command('python3 vim.vars["set_by_python3"] = [100, 0]')
    eq({100, 0}, eval('g:set_by_python3'))
  end)

  it('python3_execute with nested commands', function()
    command([[python3 vim.command('python3 vim.command("python3 vim.command(\'let set_by_nested_python3 = 555\')")')]])
    eq(555, eval('g:set_by_nested_python3'))
  end)

...
end)
```